plugins/outputs/postgresql/postgresql_test.go

```diff
@@ -0,0 +1,768 @@
+package postgresql
```

Comment:
The only option to be able to run most of the tests offline is with something like [pgmock](https://github.com/jackc/pgmock). There are 2 main challenges with that.
1. It would be very intensive work to set it up with all the back and forth this plugin does against the database.
2. It would be very fragile. Practically any change to the plugin would require changing the mocked calls/responses. Meaning you'd have to spin up the database to record the calls so you can mock them. It's my **opinion** that this is not worth it as it results in you needing to:
    a. spin up a postgres server, thus defeating the point of having offline tests
    b. change the tests to match the slightest change in behavior. Someone making a change is more likely to take the recorded calls and blindly apply them. By using a real database we check the end result, not the intermediate steps.